DUI3-DX/Converters/ArcGIS/Speckle.Converters.ArcGIS3/Layers/VectorLayerToSpeckleConverter.cs

```diff
@@ -24,33 +30,60 @@ public Base Convert(object target)
 
   public VectorLayer RawConvert(FeatureLayer target)
   {
-    var speckleLayer = new VectorLayer();
-    var spatialRef = target.GetSpatialReference();
+    VectorLayer speckleLayer = new();
+
+    // get document CRS (for writing geometry coords)
+    var spatialRef = _contextStack.Current.Document.SpatialReference;
     speckleLayer.crs = new CRS
     {
       wkt = spatialRef.Wkt,
       name = spatialRef.Name,
       units_native = spatialRef.Unit.ToString(),
     };
+
+    // other properties
     speckleLayer.name = target.Name;
+    speckleLayer.units = _contextStack.Current.SpeckleUnits;
 
     // get feature class fields
-    var attributes = new Base();
-    IReadOnlyList<Field> fields = target.GetTable().GetDefinition().GetFields();
-    foreach (Field field in fields)
+    var layerAttributes = new Base();
+    var dispayTable = target as IDisplayTable;
+    IReadOnlyList<FieldDescription> fieldDescriptions = dispayTable.GetFieldDescriptions();
+    foreach (FieldDescription field in fieldDescriptions)
     {
       string name = field.Name;
-      // breaks on Raster Field type when assigning indiv. GisFeature values
-      if (name != "Shape" && field.FieldType.ToString() != "Raster")
+      if (name == "Shape")
       {
-        attributes[name] = field.FieldType;
+        continue;
+      }
+      if (field.IsVisible)
+      {
+        layerAttributes[name] = field.Type;
       }
     }
-    speckleLayer.attributes = attributes;
+    speckleLayer.attributes = layerAttributes;
     speckleLayer.nativeGeomType = target.ShapeType.ToString();
 
-    // search the rows
+    // get a simple geometry type
+    string spekleGeometryType = "None";
+    if (speckleLayer.nativeGeomType.ToLower().Contains("point"))
+    {
+      spekleGeometryType = "Point";
+    }
+    else if (speckleLayer.nativeGeomType.ToLower().Contains("polyline"))
+    {
+      spekleGeometryType = "Polyline";
+    }
+    else if (speckleLayer.nativeGeomType.ToLower().Contains("polygon"))
+    {
+      spekleGeometryType = "Polygon";
+    }
+    else if (speckleLayer.nativeGeomType.ToLower().Contains("multipatch"))
+    {
+      spekleGeometryType = "Multipatch";
+    }
```

Comment:
This could be also refactored out into a private method. It also seems like this comparisons are quite weak: anything that contains `point` will match. Could we make it more specific or is this by design?

Reply:
Splitting into a separate method👍  but the simplification is intended for interop between different GIS software. Each of the classes can be either Simple- or Multi-  (and some only simple or only multi), but Speckle writes everything as Multitype by default.